packages/reputation-miner/ReputationMiner.js

```diff
@@ -1060,6 +1062,34 @@ class ReputationMiner {
     );
   }
 
+  async updatePeriodLength(repCycle) {
+    const { numerator, denominator } = await repCycle.getDecayConstant();
+
+    // Hard to do logs with bignumbers, so let's just see what happens!
+    let value = ethers.BigNumber.from("1000000000000000000");
+    // Due to rounding, and how we conventionally define the decay constant, after the
+    // intended halfing period, we will still be very slightly above half and only drop
+    // below half on the next period. This starting value takes that in to account.
+    let count = -1;
+    while (value.gt(ethers.BigNumber.from("500000000000000000"))){
+      value = value.mul(numerator).div(denominator);
+      count += 1;
+    }
+    const calculatedPeriodLength = Math.floor(90 * DAY_IN_SECONDS / count);
+
+    this.miningCycleDuration = await repCycle.getMiningWindowDuration();
+    this.miningCycleDuration = this.miningCycleDuration.toNumber();
+    this.constant = ethers.BigNumber.from(2).pow(256).sub(1).div(this.miningCycleDuration);
+
+    if (calculatedPeriodLength !== this.miningCycleDuration) {
```

Comment:
Ok, so we use the constants for the canonical window cycle, but there's still the explicit value in the contracts, which we use to throw warnings? Remind me why we don't just use the explicit value?

Reply:
I mostly just want something to be doing a sanity check on these values (which we don't currently do).